DOC-12193 Document sensitive cluster settings and the required privileges to view them (#19355)
cockroachdb/docs: src/current/v23.2/cluster-settings.md

## Change a cluster setting

Cluster settings can be updated anytime after a cluster has been started. Use the [`SET CLUSTER SETTING`]({% link {{ page.version.version }}/set-cluster-setting.md %}) statement.

Before changing a cluster setting, note the following:

- 	Changing a cluster setting is not instantaneous, as the change must be propagated to other nodes in the cluster.

- 	Do not change cluster settings while [upgrading to a new version of CockroachDB]({% link {{ page.version.version }}/upgrade-cockroach-version.md %}). Wait until all nodes have been upgraded before you make the change.

## Sensitive settings

{% include_cached new-in.html version="v23.2.1" %} Due to security concerns, some cluster settings are considered sensitive. Their values should not be visible to all users when using the [`SHOW CLUSTER SETTING`]({% link {{ page.version.version }}/show-cluster-setting.md %}) statement.

The values of these sensitive settings are always visible to users with the `admin` role or the `MODIFYCLUSTERSETTING` privilege.

> The values of these sensitive settings are also visible to users with the `VIEWCLUSTERSETTING` privilege when [`server.redact_sensitive_settings.enabled`]({% link {{ page.version.version }}/cluster-settings.md %}#setting-server-redact-sensitive-settings-enabled) is disabled.

Reviewer: I would consider turning this into two bullets, because on the page it can be strange to read the same sentence start twice. So, "The values of these sensitive settings are visible to users with: 
- The `admin` role or the `MODIFYCLUSTERSETTING` privilege.
- The `VIEWCLUSTERSETTING` privilege when [`server.redact_sensitive_settings.enabled`]({% link {{ page.version.version }}/cluster-settings.md %}#setting-server-redact-sensitive-settings-enabled) is disabled."

> {% include_cached new-in.html version="v23.2.1" %} Due to security concerns, some cluster settings are considered sensitive. Their values should not be visible to all users when using the [`SHOW CLUSTER SETTING`]({% link {{ page.version.version }}/show-cluster-setting.md %}) statement.
> 
> The values of these sensitive settings are always visible to users with the `admin` role or the `MODIFYCLUSTERSETTING` privilege.
> 
> The values of these sensitive settings are also visible to users with the `VIEWCLUSTERSETTING` privilege when [`server.redact_sensitive_settings.enabled`]({% link {{ page.version.version }}/cluster-settings.md %}#setting-server-redact-sensitive-settings-enabled) is disabled.
> 
> To enable redaction of these sensitive setting values for non-`admin` users without the `MODIFYCLUSTERSETTING` privilege, such as users with the `VIEWCLUSTERSETTING` privilege, run:

Reviewer 2: I wouldn't frame as "Their values should not be visible to all users", because we default to having the redaction disabled. Attempting to change this led me down a path where I ended up suggesting a number of changes; feel free to take/leave!

```suggestion
{% include_cached new-in.html version="v23.2.1" %} You can prevent users without sufficient permissions from viewing the values of cluster settings that CockroachDB classifies as sensitive.

By default, users with the `VIEWCLUSTERSETTING` privilege can view the values of all settings displayed when using the [`SHOW CLUSTER SETTING`]({% link {{ page.version.version }}/show-cluster-setting.md %}) statement.

If you enable the option to redact sensitive settings, the sensitive setting values are hidden from those users, and visible only to users with the `admin` role or the `MODIFYCLUSTERSETTING` privilege.

To enable this redaction of sensitive setting values, set the cluster setting [`server.redact_sensitive_settings.enabled`]({% link {{ page.version.version }}/cluster-settings.md %}#setting-server-redact-sensitive-settings-enabled) to `true`:
```

{% include_cached copy-clipboard.html %}
~~~ sql
SET CLUSTER SETTING server.redact_sensitive_settings.enabled = 'true';
~~~

> The table below summarizes when sensitive settings values are visible or redacted:

Reviewer: ```suggestion
The table summarizes when sensitive settings values are visible or redacted:
```

> | User attribute | Redaction disabled | Redaction enabled |

Reviewer 2: Consider adding a 4th row like

`None of the above roles/privileges | not visible | not visible1

|----------------|:------------------:|:-----------------:|
|`admin` role | visible | visible
|`MODIFYCLUSTERSETTING` [system-level privilege]({% link {{ page.version.version }}/security-reference/authorization.md %}#privileges) | visible | visible

> |`VIEWCLUSTERSETTING` [system-level privilege]({% link {{ page.version.version }}/security-reference/authorization.md %}#privileges) | visible | **redacted**

Reviewer 2: ```suggestion
|`VIEWCLUSTERSETTING` [system-level privilege]({% link {{ page.version.version }}/security-reference/authorization.md %}#privileges) | visible | **redacted**
```

We currently have a style issue with bold in tables, apparently. I'd suggest un-bolding for now.
